itest/litd_firewall_test.go

```diff
@@ -155,24 +161,492 @@ func testFWRateLimitAndPrivacyMapper(net *NetworkHarness, t *harnessTest) {
+	// channel IDs of the restricted channels have not changed.
+	aliceObfchans := make(map[uint64]bool)
+	for _, c := range chansToRestrict {
+		oc := getPseudo(sessResp.Session.GroupId, c, "")
+		ocInt, err := firewalldb.StrToUint64(oc)
+		require.NoError(t.t, err)
+		require.True(t.t, obfuscatedChansToRestrict[ocInt])
+
+		aliceObfchans[ocInt] = true
+	}
+
+	// This time, there should also be an entry for the unrestricted
+	// channel.
+	obfuscatedUnrestrictedChan := getPseudo(
+		sessResp.Session.GroupId, unrestrictedChan, "",
+	)
+	obfUnrestrictedChanID, err := firewalldb.StrToUint64(
+		obfuscatedUnrestrictedChan,
+	)
+	require.NoError(t.t, err)
+
+	aliceObfchans[obfUnrestrictedChanID] = true
+
+	// Iterate over the channels list sent to the autopilot server in the
+	// fee report and ensure that they match the set of obfuscated channel
+	// IDs we have for Alice. Also use this opportunity to get the pseudo
+	// string representing the un-restricted channel point so that we can
+	// make a write call on it later.
+	var obfUnrestrictedChanPoint string
+	for _, channel := range feeReport.ChannelFees {
+		require.True(t.t, aliceObfchans[channel.ChanId])
+
+		if channel.ChanId == obfUnrestrictedChanID {
+			obfUnrestrictedChanPoint = channel.ChannelPoint
+		}
+	}
+
+	// Now we will perform a single write call from the autopilot server on
+	// the unrestricted channel.
+	pseudoTxid, pseudoIndex, err := decodeChannelPoint(
+		obfUnrestrictedChanPoint,
+	)
+	require.NoError(t.t, err)
+
+	chanPoint := &lnrpc.ChannelPoint{
+		FundingTxid: &lnrpc.ChannelPoint_FundingTxidStr{
+			FundingTxidStr: pseudoTxid,
+		},
+		OutputIndex: pseudoIndex,
+	}
+	_, err = lndConn.UpdateChannelPolicy(
+		ctx, &lnrpc.PolicyUpdateRequest{
+			BaseFeeMsat: 9,
+			FeeRate:     8,
+			Scope: &lnrpc.PolicyUpdateRequest_ChanPoint{
+				ChanPoint: chanPoint,
+			},
+			TimeLockDelta: 20,
+			MaxHtlcMsat:   100000,
+		}, caveatCreds1,
+	)
+	require.NoError(t.t, err)
+
+	// Try to create a new session linked to the previous one. This should
+	// fail due to the previous one still being active.
+	_, err = litAutopilotClient.AddAutopilotSession(
+		ctxm, &litrpc.AddAutopilotSessionRequest{
+			Label: "integration-test",
+			ExpiryTimestampSeconds: uint64(
+				time.Now().Add(5 * time.Minute).Unix(),
+			),
+			MailboxServerAddr: mailboxServerAddr,
+			Features:          sessFeatures,
+			LinkedGroupId:     sessResp.Session.GroupId,
+		},
+	)
+	require.ErrorContains(t.t, err, "is still active")
+
+	// Revoke the previous one and repeat.
+	_, err = litAutopilotClient.RevokeAutopilotSession(
+		ctxm, &litrpc.RevokeAutopilotSessionRequest{
+			LocalPublicKey: sessResp.Session.LocalPublicKey,
+		},
+	)
+	require.NoError(t.t, err)
+
+	// Also close the autopilot connection to the first session so that it
+	// doesn't continue to try and connect.
+	require.NoError(t.t, pilotConn.Close())
+	pilotConn = nil
+
+	sessResp2, err := litAutopilotClient.AddAutopilotSession(
```

Comment:
Would we clarify the linkage even further if we before we open this session:
1. Open a new channel.
2. Show that after opening the new session here, `getPseudo` get's "no such key found" for that ChanPoint, using the "original" `sessResp.Session.GroupId` (even though it's the same for all linked sessions).
3. Then at the end of the function after the `FeeReport` has been made with the new `lndConn`, an entry for that new chanPoint exists in the privacy map DB, and is found through `getPseudo` with the "original" `sessResp.Session.GroupId`.

Do you think that would add value?

Reply:
in not sure if this adds anything tbh - cause we already call `FeeReport` with the second connection and assert that it is receiving the same result as the first one. So this already proves that the same privacy map is being used for the two sessions

Reply:
Fair enough, good point, agree :)!